### PR TITLE
update provisioning state info

### PR DESCRIPTION
**Pull request by zhouhao3**
Signed-off-by: Zhou Hao <zhouhao@fujitsu.com>

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -391,18 +391,22 @@ Settings related to deploying an image to the host.
 * *state* -- The current state of any ongoing provisioning operation.
   The following are the currently supported ones:
   * *\<empty string\>* -- There is no provisioning happening, at the moment.
+  * *unmanaged* -- There is insufficient information available to register the host.
```
Comment (furkatgofurov7) [inline docs/api.md:394]:
```suggestion
  * *unmanaged* -- There is an insufficient information available to register the host.
```

File: docs/api.md [hunk 2]
```diff
@@ -391,18 +391,22 @@ Settings related to deploying an image to the host.
...
+  * *preparing* -- The existing configuration will be removed, and the new
+    configuration will be set on the host.
   * *provisioning* -- An image is being written to the host's disk(s).
   * *provisioned* -- An image has been completely written to the host's
     disk(s).
   * *externally provisioned* -- Metal³ does not manage the image on the host.
   * *deprovisioning* -- The image is being wiped from the host's disk(s).
   * *inspecting* -- The hardware details for the host are being collected
     by an agent.
+  * *deleting* -- The host ready for deletion is cleaning up.
```
Comment (furkatgofurov7) [inline docs/api.md:409]: maybe:
```suggestion
  * *deleting* -- The host is being deleted from the cluster.
```

File: docs/api.md (hunk 1 above)
Comment (zhouhao3) [reply docs/api.md]: updated, PTAL.

**Review by fmuyassarov — COMMENTED** [review]
/lgtm

**Pull request merged**